devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -0,0 +1,672 @@
+	}
+
+    simulator { }
+
+    preferences {
+        configParams.each {
+            if (it.name) {
+                if (it.range) {
+                    getNumberInput(it)
+                }
+                else {
+                    getOptionsInput(it)
+                }
+            }
+        }
+    }
+}
+
+private getOptionsInput(param) {
+    input "configParam${param.num}", "enum",
+            title: "${param.name}:",
+            required: false,
+            defaultValue: "${param.value}",
+            options: param.options
+}
+
+private getNumberInput(param) {
+    input "configParam${param.num}", "number",
+            title: "${param.name}:",
+            required: false,
+            defaultValue: "${param.value}",
+            range: param.range
+}
+
+def installed() {
+    logDebug "installed()..."
+    state.refreshConfig = true
+}
+
+def updated() {
+    if (!isDuplicateCommand(state.lastUpdated, 5000)) {
+        state.lastUpdated = new Date().time
+
+        logDebug "updated()..."
+
+        initialize()
+
+        runIn(5, executeConfigureCmds, [overwrite: true])
+    }
+
+    return []
+}
+
+private initialize() {
+    def checkInterval = ((60 * 60 * 3) + (5 * 60))
+
+    def checkIntervalEvt = [name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"]]
+
+    if (!device.currentValue("checkInterval")) {
+        sendEvent(checkIntervalEvt)
+    }
+}
+
+
+def configure() {
+    logDebug "configure()..."
+
+    if (state.resyncAll == null) {
+        state.resyncAll = true
+        runIn(8, executeConfigureCmds, [overwrite: true])
+    }
+    else {
+        if (!pendingChanges) {
+            state.resyncAll = true
+        }
+        executeConfigureCmds()
+    }
+    return []
+}
+
+def executeConfigureCmds() {
+    runIn(6, refreshSyncStatus)
+
+    def cmds = []
+
+//	cmds << sensorBinaryGetCmd()
+//	cmds << batteryGetCmd()
+
+    configParams.each { param ->
+        def storedVal = getParamStoredValue(param.num)
+        def paramVal = param.value
+        if (state.resyncAll || ("${storedVal}" != "${paramVal}")) {
+            cmds << configSetCmd(param, paramVal)
+            if (param.num == temperatureReportThresholdParam.num) {
+                cmds << "delay 3000"
+                cmds << sensorMultilevelGetCmd(tempSensorType)
+            }else{
+                cmds << configGetCmd(param)
+            }
+        }
+    }
+
+	state.resyncAll = false
+	if (cmds) {
+		sendCommands(delayBetween(cmds, 500))
+	}
+	return []
+}
+
+private sendCommands(cmds) {
+    def actions = []
+    cmds?.each {
+        actions << new physicalgraph.device.HubAction(it)
+    }
+    sendHubCommand(actions, 100)
+    return []
+}
+
+
+// Required for HealthCheck Capability, but doesn't actually do anything because this device sleeps.
+def ping() {
+	logDebug "ping()"	
+}
```

Comment:
...implement proper `ping()` method which will be able to bring device back from offline state.

Reply:
OK, I removed the line 236.